aws-lc-rs/src/rsa/key.rs

```diff
@@ -106,24 +98,28 @@ pub struct KeyPair {
     // other thread is concurrently calling a mutating function. Unless otherwise
     // documented, functions which take a |const| pointer are non-mutating and
     // functions which take a non-|const| pointer are mutating.
-    evp_pkey: LcPtr<EVP_PKEY>,
-    serialized_public_key: PublicKey,
+    pub(super) rsa_evp_pkey: RsaEvpPkey,
+    pub(super) serialized_public_key: PublicKey,
 }
 
 impl Sealed for KeyPair {}
 unsafe impl Send for KeyPair {}
 unsafe impl Sync for KeyPair {}
 
 impl KeyPair {
-    fn new(evp_pkey: LcPtr<EVP_PKEY>) -> Result<Self, KeyRejected> {
-        unsafe {
-            let rsa_key = evp_pkey.get_rsa()?;
-            let serialized_public_key = PublicKey::new(&rsa_key.as_const())?;
-            Ok(KeyPair {
-                evp_pkey,
-                serialized_public_key,
-            })
-        }
+    unsafe fn new(evp_pkey: LcPtr<EVP_PKEY>) -> Result<Self, KeyRejected> {
```

Comment:
I don't think this function needs to be `unsafe`.  Remove here, and in `generate` below.